### PR TITLE
sdjournal: add test for incorrect Read() length

### PR DESCRIPTION
**Pull request by lucab**
Test for the whole `JournalReader.Read()`, and in particular for incorrect read length.

Ref #172
Ref #173

### CODE REVIEW

File: sdjournal/journal_test.go [hunk 1]
```diff
@@ -228,3 +230,57 @@ func TestJournalGetEntry(t *testing.T) {
...
+	if err != nil {
+		t.Fatal("Error writing to journal", err)
+	}
+	time.Sleep(time.Duration(1) * time.Second)
+
+	// ... and try to read it back piece by piece via a small buffer
+	finalBuff := new(bytes.Buffer)
+	var e error = nil
+	var c int = -1
+	for c != 0 && e == nil {
```
Comment (jonboulle) [inline sdjournal/journal_test.go]: for c := -1; c != 0 && e == nil; {

File: sdjournal/journal_test.go [hunk 2]
```diff
@@ -228,3 +230,57 @@ func TestJournalGetEntry(t *testing.T) {
...
+		t.Fatal("Got a nil reader")
+	}
+	defer r.Close()
+
+	want := fmt.Sprintf("%slongentry %s%s", delim, longEntry, delim)
+	err = journal.Send(want, journal.PriInfo, map[string]string{matchField: matchValue})
+	if err != nil {
+		t.Fatal("Error writing to journal", err)
+	}
+	time.Sleep(time.Duration(1) * time.Second)
```
Comment (jonboulle) [inline sdjournal/journal_test.go]: time.Sleep(time.Second)
(if we really must :-()

**Comment by jonboulle** [comment]
LGTM

**Comment by lucab** [comment]
Amended. Thanks for all the idiomatic hints.

**Pull request merged**